### DISCUSSION TITLE
When JAX solves the eigenvalue problem, if there are multiple eigenvalues, can it still obtain accurate gradients?

### DISCUSSION THREAD

**Discussion by DoTulip** [jax-ml/jax#9907] [General] [answered]
Hello, JAX Team,
I have a question about the automatic differentiation of eigenvalue problems. When JAX solves the eigenvalue problem if there are multiple eigenvalues, can it still obtain accurate gradients?

**Comment by mattjj** [comment, marked as answer]
Check out the discussion on #669 and @shoyer's comment [here](https://github.com/google/jax/issues/8732#issuecomment-982870019) on #8732 (also #2311). See also [this comment in the code](https://github.com/google/jax/blob/main/jax/_src/lax/linalg.py#L527-L534).

I think the short answer is no. Moreover I think that with repeated eigenvalues the gradient for eigendecomposition is not well defined mathematically (though for autodiff of an eig/eigh routine one could _define_ the autodiff gradient with reference to whatever gauge-symmetry-breaking choices the implementation happened to make).

@shoyer @levskaya please correct any mistakes in the above!

**Reply by DoTulip** [reply]
Thank you so much for your reply, these discussions have helped me a lot!
